webrtc/stun/stun_transport.nim

```diff
@@ -0,0 +1,97 @@
+# Nim-WebRTC
+# Copyright (c) 2024 Status Research & Development GmbH
+# Licensed under either of
+#  * Apache License, version 2.0, ([LICENSE-APACHE](LICENSE-APACHE))
+#  * MIT license ([LICENSE-MIT](LICENSE-MIT))
+# at your option.
+# This file may not be copied, modified, or distributed except according to
+# those terms.
+
+import tables
+import chronos, chronicles, bearssl
+import stun_connection, stun_protocol, ../udp_connection
+
+logScope:
+  topics = "webrtc stun stun_transport"
+
+type
+  Stun* = ref object
+    connections: Table[TransportAddress, StunConn]
+    pendingConn: AsyncQueue[StunConn]
+    readingLoop: Future[void]
+    conn: UdpConn
+    iceTiebreaker: uint64
+    rng: ref HmacDrbgContext
+
+proc accept*(self: Stun): Future[StunConn] {.async: (raises: [CancelledError]).} =
+  ## Accept a Stun Connection
+  ##
+  var res: StunConn
+  while true:
+    res = await self.pendingConn.popFirst()
+    if res.closed != true: # The connection could be closed before being accepted
+      break
+  return res
+
+proc connect*(
+    self: Stun,
+    raddr: TransportAddress
+  ): Future[StunConn] {.async: (raises: []).} =
+  ## Connect to a remote address, creating a Stun Connection
+  ##
+  if self.connections.hasKey(raddr):
+    try:
+      return self.connections[raddr]
+    except KeyError as exc:
+      doAssert(false, "Should never happen")
```

Comment:
I strongly discourage this pattern and raising defects in the codebase in general, unless it is a severe problem when we do want to crash as soon as possible. That was the reason `withValue` was created in nim-libp2p.

Reply:
Done.